### ISSUE TITLE
Updating graphQL changes when you have a nested file structure

### ISSUE THREAD

**Issue by alexhawkins** [lvarayut/relay-fullstack#10]
[My current directory structure](https://www.dropbox.com/s/bp4z87683ohyaud/Screenshot%202016-04-06%2014.30.39.png?dl=0) ( see this first to understand how i have things set up )
Changes to schema.js work fine but I don't really make changes there.
[My current schema.js file](https://www.dropbox.com/s/a6lw171mkxfwtem/Screenshot%202016-04-06%2014.33.24.png?dl=0)
Gaze recognizes changes in other files but the schema.js is only updated when the schema.js file is changed. I'm using:

```
gaze(path.join(__dirname, 'graphql/**/*.js'), (err, watcher) => {
    if (err) logger.error(chalk.bold.red('Error: Watching files in data folder'));
    watcher.on('all', async() => {
      try {
        // Close the GraphQL server, update the schema.json and schema.graphql, and start the server again
        graphQLServer.close();
        await updateSchema();
        const newSchema = requireUncached(path.join(__dirname, './graphql/schema')).default;
        startGraphQLServer(newSchema);
        // Close the Relay server, and start it again
        relayServer.listeningApp.close();
        startRelayServer();
      } catch (e) {
        logger.error(chalk.bold.red((e.stack)));
      }
    });
  });

```

For example, if I make a change in UserType.js (see above image for directory structure) the schema.js is not updated with the change.

Any ideas on how to handle this? I'd like everything to hot reload on any change. Currently I'm having to stop and restart the server which is pretty time consuming. Help would be much appreciated.

**Comment by lvarayut** [comment]
It's the problem of caching files, the first time you require the UserType.js, the file will be cached by NodeJs. So, it's always the same even though it has been changed.  For a quick fix, you always have to use `requireUncached()` function to require files. 

I have planned to move from Gaze to Nodemon very soon. This will help reloading the server without worrying about caching files.

**Comment by lvarayut** [comment]
Fixed by 48a533c.

**Comment by alexhawkins** [comment]
@lvarayut Thank you very much :)
